implementation/leprechaun/src/vehicle/test/vehicle.controller.spec.ts

```diff
@@ -0,0 +1,144 @@
+import { INestApplication } from '@nestjs/common';
+import { Test, TestingModule } from '@nestjs/testing';
+import { MockFunctionMetadata, ModuleMocker } from 'jest-mock';
+import * as request from 'supertest';
+import { v7 as uuid } from 'uuid';
+import { HttpExceptionFilter, InternalExceptionFilter } from '../../shared/api/http-exception.filter';
+import { CreateVehicleDto } from '../controller/vehicle-create.dto';
+import { VehicleController } from '../controller/vehicle.controller';
+import { Vehicle } from '../database/vehicle.entity';
+import { VehicleService } from '../service/vehicle.service';
+
+const moduleMocker = new ModuleMocker(global);
+
+const UUID_SAMPLE = '019411be-c6c3-7f67-83c1-c088d4280102';
+const VEHICLE = { id: UUID_SAMPLE, sideNumber: '2136', isActive: true };
+const VEHICLES: Vehicle[] = [VEHICLE];
+
+describe('VehicleController', () => {
+  let controller: VehicleController;
+  let app: INestApplication;
+
+  afterEach(async () => {
+    await app.close();
+  });
+
+  beforeEach(async () => {
+    const module: TestingModule = await Test.createTestingModule({
+      controllers: [VehicleController],
+    })
+      .useMocker((token) => {
+        if (token === VehicleService) {
+          const mocked: {
+            [key in keyof VehicleService]: jest.Mock<VehicleService[key], Parameters<VehicleService[key]>>;
+          } = {
+            listAll: jest.fn().mockResolvedValue({ vehicles: VEHICLES, total: VEHICLES.length }),
+            findVehicleById: jest.fn().mockResolvedValue(VEHICLES[0]),
+            createVehicle: jest.fn().mockImplementation((newVehicle: CreateVehicleDto) => {
+              const vehicle = { id: uuid(), ...newVehicle, isActive: true };
+              VEHICLES.push(vehicle);
+              return vehicle;
+            }),
+            deleteVehicleById: jest.fn().mockResolvedValue(undefined),
+            updateVehicleById: jest.fn().mockResolvedValue(VEHICLES[0]),
+          };
+          return mocked;
+        }
+        if (typeof token === 'function') {
+          const mockMetadata = moduleMocker.getMetadata(token) as MockFunctionMetadata<any, any>;
```

Comment:
`<any, any>` 😠

Reply:
I have no idea what this piece of code even does. And it doesn't break any tests when I remove it!

Reply:
Great 👌